clcache.py

```diff
@@ -1612,19 +1618,28 @@ def processCompileRequest(cache, compiler, args):
     printOutAndErr(out, err)
     return exitCode
 
+def formBaseCommandLine(cmdLine, sourceFiles):
+    # type: (List[str], List[Tuple[str, str]]) -> List[str]
+    setOfSources = set([sourceFile for sourceFile, _ in sourceFiles])
+    skippedArgs = ('/MP', '/Tc', '/Tp')
+    baseCmdLine = [
+        arg for arg in cmdLine
+        if not (arg in setOfSources or arg.startswith(skippedArgs))
+    ]
+
+    return baseCmdLine
+
 def scheduleJobs(cache, compiler, cmdLine, environment, sourceFiles, objectFiles):
-    baseCmdLine = []
-    setOfSources = set(sourceFiles)
-    for arg in cmdLine:
-        if not (arg in setOfSources or arg.startswith("/MP")):
-            baseCmdLine.append(arg)
+    # type: (???, str, List[str], ???, List[Tuple[str, str]], List[str]) -> int
+    # Filter out all source files from the command line to form baseCmdLine
```

Comment:
I think this comment is misleading: the `formBaseCommandLine` function does not only filter out all source files, it also filters out `/MP`. However, I agree that the function _should_ filter out just the source files because then it could get the more descriptive name `filterSourceFiles` and the `/MP` could be filtered on the caller side:

```
baseCmdLine = [arg for arg in filterSourceFiles(cmdLine, sourceFiles) if arg != '/MP']
```

Reply:
Will do.